src/table.rs

```diff
@@ -117,6 +117,46 @@ impl Table {
         }
     }
 
+    /// Sort Key/Value Pairs of the table using the using the comparison function `compare`.
+    ///
+    /// The comparison function receives two key and value pairs to compare (you can sort by keys or
+    /// values or their combination as needed).
+    pub fn sort_by<F>(&mut self, compare: F)
+    where
+        F: Fn(&Key, &Item, &Key, &Item) -> std::cmp::Ordering,
```

Comment:
We should probably expose this in `TableLike` as well

Reply:
Oh boo. Doing this would cause object safety issues. Here's what the compiler tells me when I try to do that.

```
   Compiling toml_edit v0.12.0 (/Users/bryan/third-party/toml_edit)
error[E0038]: the trait `TableLike` cannot be made into an object
   --> src/item.rs:254:36
    |
254 |     pub fn as_table_like(&self) -> Option<&dyn TableLike> {
    |                                    ^^^^^^^^^^^^^^^^^^^^^^ `TableLike` cannot be made into an object
    |
note: for a trait to be "object safe" it needs to allow building a vtable to allow the call to be resolvable dynamically; for more information visit <https://doc.rust-lang.org/reference/items/traits.html#object-safety>
   --> src/table.rs:486:8
    |
447 | pub trait TableLike: crate::private::Sealed {
    |           --------- this trait cannot be made into an object...
...
486 |     fn sort_values_by<F>(&mut self, compare: F)
    |        ^^^^^^^^^^^^^^ ...because method `sort_values_by` has generic type parameters
    = help: consider moving `sort_values_by` to another trait
```

Reply:
Good point.  Not saying we have to do this but one option is
```rust
pub type FnCmp<'a> = dyn Fn(&Key, &Item, &Key, &Item) -> std::cmp::Ordering + 'a;
pub(crate) type FnCmpInternal<'a> = dyn Fn(&InternalString, &TableKeyValue, &InternalString, &TableKeyValue) -> std::cmp::Ordering
    + 'a;

...

    pub fn sort_values_by<'a>(&mut self, compare: &FnCmp<'a>) {
    }

    fn sort_values_by_internal<'a>(&mut self, compare: &FnCmpInternal<'a>) {
    }

```